shared/js/content-scope/canvas-protection.js

```diff
@@ -1,13 +1,20 @@
-import { shouldExemptMethod, DDGProxy, DDGReflect } from './utils'
+import { shouldExemptMethod, DDGProxy, DDGReflect, postDebugMessage } from './utils'
 import { computeOffScreenCanvas } from './canvas'
 
 export function init (args) {
-    const { sessionKey, site } = args
+    const { sessionKey, site, debug } = args
     const domainKey = site.domain
 
     // Using proxies here to swallow calls to toString etc
     const getImageDataProxy = new DDGProxy(CanvasRenderingContext2D.prototype, 'getImageData', {
         apply (target, thisArg, args) {
+            if (debug) {
+                postDebugMessage('canvas', {
```

Comment:
Could this be contained in `shouldExemptMethod` that way all implementations should get it and it allow us to toggle this too? Happy if we need to rename the method to explain the debugging.

The performance cost shouldn't be any different.